File: core/src/main/java/br/ufs/nubankchallenge/core/infrastructure/models/SubmitChargebackBody.java
package br.ufs.nubankchallenge.core.infrastructure.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by @ubiratanfsoares
 */

public class SubmitChargebackBody {

    public String comment;
    @SerializedName("reason_details") public List<ReasonDetailBody> details;

    public static class ReasonDetailBody {

        public String id;
        public boolean response;

        public ReasonDetailBody(String id, boolean response) {
            this.id = id;
            this.response = response;
        }
    }

}

File: core/src/main/java/br/ufs/nubankchallenge/core/infrastructure/models/SubmitChargebackBody.kt
package br.ufs.nubankchallenge.core.infrastructure.models

import com.google.gson.annotations.SerializedName

/**
 * Created by @ubiratanfsoares
 */

class SubmitChargebackBody {

    var comment: String? = null
    @SerializedName("reason_details") var details: List<ReasonDetailBody>? = null

    class ReasonDetailBody(var id: String, var response: Boolean)

}
